Sources/API/Converter.swift

```diff
@@ -48,4 +48,990 @@ enum Converter {
+//
+//            pbTextNodes.push(pbTextNode);
+//        }
+//
+//        return pbTextNodes;
+//    }
+    
+    /**
+     * `fromTextNode` converts the given Protobuf format to model format.
+     */
+//    function fromTextNode(pbTextNode: PbTextNode): RGATreeSplitNode<string> {
+//        const textNode = RGATreeSplitNode.create(
+//            fromTextNodeID(pbTextNode.getId()!),
+//            pbTextNode.getValue(),
+//        );
+//        textNode.remove(fromTimeTicket(pbTextNode.getRemovedAt()));
+//        return textNode;
+//    }
+}
+
+// MARK: RichTextNode
+extension Converter {
+    /**
+     * `fromRichTextNode` converts the given Protobuf format to model format.
+     */
+    // TODO: RGATreeSplitNode is not implemented
+//    static func fromRichTextNode(_ pbTextNode: PbRichTextNode) -> RGATreeSplitNode<RichTextValue> {
+//        const richTextValue = RichTextValue.create(pbTextNode.getValue());
+//        pbTextNode.getAttributesMap().forEach((value) => {
+//            richTextValue.setAttr(
+//                value.getKey(),
+//                value.getValue(),
+//                fromTimeTicket(value.getUpdatedAt())!,
+//            );
+//        });
+//
+//        const textNode = RGATreeSplitNode.create(
+//            fromTextNodeID(pbTextNode.getId()!),
+//            richTextValue,
+//        );
+//        textNode.remove(fromTimeTicket(pbTextNode.getRemovedAt()));
+//        return textNode;
+//    }
+    
+    /**
+     * `fromRichText` converts the given Protobuf format to model format.
+     */
+//    function fromRichText<A>(pbText: PbJSONElement.RichText): CRDTRichText<A> {
+//        const rgaTreeSplit = new RGATreeSplit<RichTextValue>();
+//
+//        let prev = rgaTreeSplit.getHead();
+//        for (const pbNode of pbText.getNodesList()) {
+//            const current = rgaTreeSplit.insertAfter(prev, fromRichTextNode(pbNode));
+//            if (pbNode.hasInsPrevId()) {
+//                current.setInsPrev(
+//                    rgaTreeSplit.findNode(fromTextNodeID(pbNode.getInsPrevId()!)),
+//                );
+//            }
+//            prev = current;
+//        }
+//        const text = new CRDTRichText<A>(
+//            rgaTreeSplit,
+//            fromTimeTicket(pbText.getCreatedAt())!,
+//        );
+//        text.setMovedAt(fromTimeTicket(pbText.getMovedAt()));
+//        text.setRemovedAt(fromTimeTicket(pbText.getRemovedAt()));
+//        return text;
+//    }
+}
+
+// MARK: ChangePack
+extension Converter {
+    /**
+     * `toChangePack` converts the given model to Protobuf format.
+     */
+    static func toChangePack(pack: ChangePack) -> PbChangePack {
+        var pbChangePack = PbChangePack();
+        pbChangePack.documentKey = pack.getDocumentKey()
+        pbChangePack.checkpoint = toCheckpoint(pack.getCheckpoint())
+        pbChangePack.changes = toChanges(pack.getChanges())
+        // TODO: snapshot can be nil!
+        pbChangePack.snapshot = pack.getSnapshot() ?? Data()
```

Comment:
'snapshot' of ChangePack is optional, but PbChangePack is required. 
We need the default value of snapshot or change PbChangePack's to optional. 

@hackerwins

Reply:
Handle empty bytes as nil.